File: examples/python-examples/MeshModification.dox.py
import math

import meshlib.mrmeshpy as mrmeshpy

mesh = mrmeshpy.makeTorus()

# Relax mesh (5 iterations)
relax_params = mrmeshpy.MeshRelaxParams()
relax_params.iterations = 5
mrmeshpy.relax(mesh, relax_params)

# Subdivide mesh
props = mrmeshpy.SubdivideSettings()
props.maxDeviationAfterFlip = 0.5
mrmeshpy.subdivideMesh(mesh, props)

# Rotate mesh
rotation_xf = mrmeshpy.AffineXf3f.linear(mrmeshpy.Matrix3f.rotation(mrmeshpy.Vector3f.plusZ(), math.pi * 0.5))
mesh.transform(rotation_xf)

File: examples/js-examples/MeshModification.dox.js
import createMeshLib from '@meshinspector/meshlib';

const ml = await createMeshLib();

using mesh = ml.makeTorus(1.0, 0.1, 16, 16);

// Relax mesh (5 iterations)
using relaxParams = new ml.MeshRelaxParams();
relaxParams.iterations = 5;
ml.relax(mesh, relaxParams);

// Subdivide mesh
using subdivideSettings = new ml.SubdivideSettings();
subdivideSettings.maxDeviationAfterFlip = 0.5;
ml.subdivideMesh(mesh, subdivideSettings);

// Rotate mesh
using rotation = ml.Matrix3f.rotation({ x: 0, y: 0, z: 1 }, Math.PI * 0.5);
using rotationXf = ml.AffineXf3f.linear(rotation);
mesh.transform(rotationXf);
